pyxem/signals/electron_diffraction.py

```diff
@@ -768,10 +768,12 @@ def find_peaks(self, method='skimage', *args, **kwargs):
         peaks.map(peaks_as_gvectors,
                   center=np.array(self.axes_manager.signal_shape)/2,
                   calibration=self.axes_manager.signal_axes[0].scale)
+        #TODO: This class reassignment is brittle and a little complicated to
+        #correct. DiffractionVectors needs to be ok with being ragged or not
+        #currently difficult to reconcile... 
         peaks = DiffractionVectors(peaks)
```

Comment:
This peaks assignments gets moved (to the line above the return statement) in the pull request coming in, the methods on the lines below it currently convert a DiffractionVector object back to a Signal2D. Not sure if it solves the problem but worth noting.

Reply:
I think it depends on what map() returns - if it's ragged I think the implementation here works fine, if it's not ragged though it does what you say.

Today I haven't been able to get something that works in both cases and doesn't break other methods, but it wasn't what I was focusing on.

Reply:
Do we have tests yet for this yet? I might do that (if no tests currently) + resolve the conflicts now.

Reply:
I've got a notebook which does a good deal of testing - I'll send it to you by e-mail...